packages/retail-ui-extensions/src/extension-api/scanner-api/scanner-api.ts

```diff
@@ -0,0 +1,32 @@
+import type {RemoteSubscribable} from '@remote-ui/async-subscription';
+
+/** The scanner source the POS device supports. */
+export type ScannerSource = 'camera' | 'external' | 'embedded';
+
+export interface ScannerSubscriptionResult {
+  /** The string data from the last scanner event received. */
+  data: string;
+  /** The scanning source from which the scan event came. */
+  source: ScannerSource;
+}
+
+export interface ScannerApiContent {
+  /** Creates a subscription to scan events
+   * Provides an initial value and a callback to subsribe to value changes. Currently supports only one subscription.
+   * You can utilize `makeStatefulSubscribable` on a `RemoteSubscribable` to implement multiple subscriptions.
+   * Using `makeStatefulSubscribable` or the corresponding hooks counts as a subscription.
+   */
+  scannerDataSubscribable: RemoteSubscribable<
+    ScannerSubscriptionResult | undefined
```

Comment:
@heltisace @vctrchu @NathanJolly Since the entire object is now the result of a scan, what do you think about this instead? The subscribable returns undefined if there hasnt been a scan yet. data and source are now both mandatory values

Reply:
I'm good with that but I have a question about the name. The Result contains both `data` and `source`. Can we name this variable and the hooks somehow better?

Reply:
Fixed it here https://github.com/Shopify/ui-extensions/pull/888